x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_pre_packaged_rules.tsx

```diff
@@ -83,223 +70,61 @@ export const usePrePackagedRules = ({
   hasEncryptionKey,
   isSignalIndexExists,
 }: UsePrePackagedRuleProps): ReturnPrePackagedRulesAndTimelines => {
-  const [prepackagedDataStatus, setPrepackagedDataStatus] = useState<
-    Pick<
-      ReturnPrePackagedRulesAndTimelines,
-      | 'createPrePackagedRules'
-      | 'refetchPrePackagedRulesStatus'
-      | 'rulesCustomInstalled'
-      | 'rulesInstalled'
-      | 'rulesNotInstalled'
-      | 'rulesNotUpdated'
-      | 'timelinesInstalled'
-      | 'timelinesNotInstalled'
-      | 'timelinesNotUpdated'
-    >
-  >({
-    createPrePackagedRules: null,
-    refetchPrePackagedRulesStatus: null,
-    rulesCustomInstalled: null,
-    rulesInstalled: null,
-    rulesNotInstalled: null,
-    rulesNotUpdated: null,
-    timelinesInstalled: null,
-    timelinesNotInstalled: null,
-    timelinesNotUpdated: null,
-  });
-
-  const [loadingCreatePrePackagedRules, setLoadingCreatePrePackagedRules] = useState(false);
-  const [loading, setLoading] = useState(true);
-  const { addError, addSuccess } = useAppToasts();
-
-  const getSuccessToastMessage = (result: {
-    rules_installed: number;
-    rules_updated: number;
-    timelines_installed: number;
-    timelines_updated: number;
-  }) => {
-    const {
-      rules_installed: rulesInstalled,
-      rules_updated: rulesUpdated,
-      timelines_installed: timelinesInstalled,
-      timelines_updated: timelinesUpdated,
-    } = result;
-    if (rulesInstalled === 0 && (timelinesInstalled > 0 || timelinesUpdated > 0)) {
-      return i18n.TIMELINE_PREPACKAGED_SUCCESS;
-    } else if ((rulesInstalled > 0 || rulesUpdated > 0) && timelinesInstalled === 0) {
-      return i18n.RULE_PREPACKAGED_SUCCESS;
-    } else {
-      return i18n.RULE_AND_TIMELINE_PREPACKAGED_SUCCESS;
+  const { data: prePackagedRulesStatus, isFetching } = usePrePackagedRulesStatus();
+  const { mutate: installPrePackagedRules, isLoading: loadingCreatePrePackagedRules } =
+    useInstallPrePackagedRules();
+
+  const createPrePackagedRules = useCallback(() => {
+    if (
+      canUserCRUD &&
+      hasIndexWrite &&
+      isAuthenticated &&
+      hasEncryptionKey &&
+      isSignalIndexExists
+    ) {
+      installPrePackagedRules();
```

Comment:
The tech debt cleanup here is fantastic @xcrzx! Like finally paying off 150 year mortgage that has been following your family for generations, haha! 😅  Thanks for tackling this one -- so much cleaner and easier to grok the actual control flow with this and `useInstallPrePackagedRules`. 💯💯💯